lib/linguist/languages.yml

```diff
@@ -1132,6 +1132,11 @@ Diff:
   codemirror_mode: diff
   codemirror_mime_type: text/x-diff
   language_id: 88
+DirectX 3D File:
+  type: data
+  extensions:
+  - ".x"
+  language_id: 201049282
```

Comment:
Alright, first, let's add the syntax highlighting fields:

~~~yaml
- ace_mode: text
- tm_scope: source.direct-x
~~~

(The [ACE editpr](https://ace.c9.io) doesn't have that many editing modes, so we usually leave this field as `text`. It's a mandatory field, however).

Reply:
what about the codemirror_* fields?

Reply:
Nope, those are optional. CodeMirror [doesn't have that many language-modes either](https://codemirror.net/mode/index.html), so unless it has one that produces similar-enough highlighting, we generally just leave these fields out (it defaults to `text/plain` in any case).